ambry-frontend/src/test/java/com.github.ambry.frontend/GetPeersHandlerTest.java

```diff
@@ -0,0 +1,393 @@
+   * @param expectedErrorCode the {@link RestServiceErrorCode} expected in response.
+   * @throws Exception
+   */
+  private void doBadArgsTest(String name, String port, RestServiceErrorCode expectedErrorCode) throws Exception {
+    StringBuilder uri = new StringBuilder(AmbryBlobStorageService.OPERATION_GET_PEERS + "?");
+    if (name != null) {
+      uri.append(GetPeersHandler.NAME_QUERY_PARAM).append("=").append(name);
+    }
+    if (port != null) {
+      if (name != null) {
+        uri.append("&");
+      }
+      uri.append(GetPeersHandler.PORT_QUERY_PARAM).append("=").append(port);
+    }
+    JSONObject data = new JSONObject();
+    data.put(MockRestRequest.REST_METHOD_KEY, RestMethod.GET);
+    data.put(MockRestRequest.URI_KEY, uri);
+    RestRequest restRequest = new MockRestRequest(data, null);
+    try {
+      sendRequestGetResponse(restRequest, new MockRestResponseChannel());
+      fail("Request should have failed");
+    } catch (RestServiceException e) {
+      assertEquals("Unexpected RestServiceErrorCode", expectedErrorCode, e.getErrorCode());
+    }
+  }
+
+  /**
+   * Implementation of {@link SecurityService} that can throw exceptions on demand.
+   */
+  private static class DummySecurityService implements SecurityService {
+    Exception exceptionToReturn = null;
+
+    @Override
+    public Future<Void> processRequest(RestRequest restRequest, Callback<Void> callback) {
+      FutureResult<Void> futureResult = new FutureResult<>();
+      futureResult.done(null, exceptionToReturn);
+      callback.onCompletion(null, exceptionToReturn);
+      return futureResult;
+    }
+
+    @Override
+    public Future<Void> processResponse(RestRequest restRequest, RestResponseChannel responseChannel, BlobInfo blobInfo,
+        Callback<Void> callback) {
+      throw new IllegalStateException();
+    }
+
+    @Override
+    public void close() throws IOException {
+
+    }
+  }
+}
+
+/**
+ * Implementation of {@link ClusterMap} that has well known structure and can provide peer lists via the
+ * {@link #getPeers(String)} method for verification.
+ * <p/>
+ * Can also throw exceptions on demand.
+ */
+class TailoredPeersClusterMap implements ClusterMap {
+  static final String[] datanodeNames = {"host1:100", "host1:200", "host2:100", "host3:100"};
+
+  private static final String DC_NAME = "Datacenter";
+  private static final String[] mountPathsArr = {"/tmp1", "/tmp2", "/tmp3", "/tmp4"};
+
+  private final Map<String, MockDataNodeId> datanodes = new HashMap<>();
+  private final Map<String, Set<String>> peerMap = new HashMap<>();
+  private final List<MockPartitionId> partitions = new ArrayList<>();
+
+  RuntimeException exceptionToThrow = null;
+
+  TailoredPeersClusterMap() {
+    for (String datanodeName : datanodeNames) {
+      String[] parts = datanodeName.split(":");
+      Port port = new Port(Integer.parseInt(parts[1]), PortType.PLAINTEXT);
+      List<String> mountPaths = Collections.singletonList("/" + datanodeName);
```

Comment:
Should this be based on `mountPathsArr`? I couldn't find any uses of this variable